Userland/Libraries/LibWeb/Layout/FormattingContext.cpp

```diff
@@ -1349,10 +1349,16 @@ void FormattingContext::compute_inset(NodeWithStyleAndBoxModelMetrics const& box
     if (box.computed_values().position() != CSS::Positioning::Relative)
         return;
 
-    auto resolve_two_opposing_insets = [&](CSS::LengthPercentage const& computed_first, CSS::LengthPercentage const& computed_second, CSSPixels& used_start, CSSPixels& used_end, CSSPixels reference_for_percentage) {
+    auto resolve_two_opposing_insets = [&](CSS::LengthPercentage const& computed_first, CSS::LengthPercentage const& computed_second, CSSPixels& used_start, CSSPixels& used_end, CSSPixels reference_for_percentage, bool containing_block_has_definite_size) {
         auto resolved_first = computed_first.to_px(box, reference_for_percentage);
         auto resolved_second = computed_second.to_px(box, reference_for_percentage);
 
+        // Avoid resolving against indefinite sizes.
+        if (computed_first.is_calculated() && computed_first.calculated()->contains_percentage() && !containing_block_has_definite_size)
+            resolved_first = 0;
+        if (computed_second.is_calculated() && computed_second.calculated()->contains_percentage() && !containing_block_has_definite_size)
+            resolved_second = 0;
```

Comment:
This feels like a very targeted fix for a more general problem.

Why doesn't `to_px()` already handle this case nicely? If the containing block has an indefinite size, I would expect the `reference_for_percentage` to be zero already(?)